src/box/xrow_update_array.c

```diff
@@ -310,6 +310,36 @@ xrow_update_array_store(struct xrow_update_field *field,
 	return out - out_begin;
 }
 
+/*
+ * Helper function that appends nils in the end so that op will insert
+ * without gaps
+ */
+int
+xrow_update_op_append_nils(struct xrow_update_op *op,
+			   struct xrow_update_field *field)
+{
+	struct xrow_update_rope *rope = field->array.rope;
+	uint32_t size = xrow_update_rope_size(rope);
+	if (op->field_no < 0 || (uint32_t)op->field_no <= size)
+		return 0;
+	uint32_t nil_count = op->field_no - size;
+	struct xrow_update_array_item *item =
+		(struct xrow_update_array_item *)
+		xrow_update_alloc(rope->ctx, sizeof(*item));
+	if (item == NULL)
+		return -1;
+	uint32_t nil_size = mp_sizeof_nil();
+	uint32_t byte_count = nil_size * nil_count;
+	char *pos = (char *) region_alloc(rope->ctx, byte_count);
+	if (pos == NULL)
+		return -1;
+	memset(pos, 0xc0, byte_count);
+	xrow_update_array_item_create(item, XUPDATE_NOP, pos, byte_count, 0);
+	if (xrow_update_rope_insert(rope, op->field_no, item, nil_count) != 0)
```

Comment:
You can do `return xrow_update_rope_insert()` instead of the `if`. It might even activate the tail-call optimization.